rest/server_context.go

```diff
@@ -815,21 +815,44 @@ func (sc *ServerContext) installPrincipals(context *db.DatabaseContext, spec map
 		} else {
 			princ.Name = &name
 		}
-		_, err := context.UpdatePrincipal(*princ, (what == "user"), isGuest)
-		if err != nil {
-			// A conflict error just means updatePrincipal didn't overwrite an existing user.
-			if status, _ := base.ErrorAsHTTPStatus(err); status != http.StatusConflict {
-				return err
+
+		worker := func() (shouldRetry bool, err error, value interface{}) {
+			_, err = context.UpdatePrincipal(*princ, (what == "user"), isGuest)
+			if err != nil {
+				if base.IsConflictError(err) {
+					// Ignore and absorb this error if it's a conflict error, which just means that updatePrincipal didn't overwrite an existing user.
+					// Since if there's an existing user it's "mission accomplished", this can be treated as a success case.
+					return false, nil, nil
+				}
+				// Assume it might be a timeout error due to view re-indexing.  So retry
+				base.LogTo("Auth", "Error calling UpdatePrincipal(): %v.  Will retry in case this is a temporary error", err)
+				return true, err, nil
 			}
-		} else if isGuest {
+
+			// No errors, assume it worked
+			return false, nil, nil
+
+		}
+
+		err, _ := base.RetryLoop("installPrincipals", worker, base.CreateDoublingSleeperFunc(16, 10))
```

Comment:
Wasn't sure what to use here.

Reply:
@adamcfraser does this look like a reasonable retry value?

`err, _ := base.RetryLoop("installPrincipals", worker, base.CreateDoublingSleeperFunc(16, 10))`

Or would it make more sense to switch away from the backoff retry and just do a normal retry?